platform/utm_mgrs_utils.cpp

```diff
@@ -110,7 +110,15 @@ UTMPoint LatLonToUtm(double lat, double lon)
   double const latTan4 = latTan2 * latTan2;
 
   int const zoneNumber = LatLonToZoneNumber(lat, lon);
-  char const zoneLetter = (lat >= 0.0) ? 'N' : 'S';
+  auto const maybeZoneLetter = LatitudeToZoneLetter(lat);
+  if (!maybeZoneLetter.has_value())
+  {
+    // Invalid latitude should be excluded by FormatUTM and FormatMGRS functions.
+    ASSERT(false, ("Invalid latitude ",lat,". Can't calculate UTM zone letter"));
+    return {};
```

Comment:
So we return just _anything_ that is not even initialized?
Probably, there is no problem here because I see:
```
  if (lat <= -80 || lat > 84)
    return {}; // Latitude limit exceeded.
```
everywhere in code.

I don't know if it is ok to have an invalid UTM for say (-88, 0), but this link gives a correct value (Z) for lat=-88
https://www.latlong.net/lat-long-utm.html

Reply:
UTM coordinates near the North and South poles are calculated with completely different formulas. For that purpose `Z` zone letter was introduced. We doesn’t support it for the moment, because those areas are not very popular. Let me update the code to cover latitude values outside of `[-80, 84)` range

Reply:
Ah, ok, no problem then, we can merge this PR now.
Then put ```ASSERT(maybeZoneLetter, (lat));``` here without ifs. Caller side makes needed checks.

Reply:
Done! Simplified UTM zone letter check.